examples/z_bytes.c

```diff
@@ -0,0 +1,189 @@
+//
+// Copyright (c) 2024 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+
+#include <ctype.h>
+#include <stddef.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <zenoh.h>
+
+#undef NDEBUG
+
+typedef struct kv_pair_t {
+    int32_t key;
+    z_owned_string_t value;
+} kv_pair_t;
+
+typedef struct kv_it {
+    kv_pair_t *current;
+    kv_pair_t *end;
+} kv_it;
+
+typedef struct int32_it {
+    int32_t *current;
+    int32_t *end;
+} int32_it;
+
+static bool kv_pairs_iter(z_owned_bytes_t *kv_pair, void *context);
+static bool int32_iter(z_owned_bytes_t *b, void *context);
+static void print_slice_data(z_view_slice_t *slice);
+
+int main(void) {
+    z_owned_bytes_t payload;
+
+    // Number types: uint8, uint16, uint32, uint64, int8, int16, int32, int64, float, double
+    uint32_t input_u32 = 123456;
+    uint32_t output_u32 = 0;
+    z_bytes_serialize_from_uint32(&payload, input_u32);
+    z_bytes_deserialize_into_uint32(z_loan(payload), &output_u32);
+    assert(input_u32 == output_u32);
+    z_drop(z_move(payload));
+    // Corresponding encoding to be used in operations options like `z_put()`, `z_get()`, etc.
+    // encoding = z_encoding_zenoh_uint32();
```

Comment:
Type missing. Shouldn't be `z_loaned_encoding_t *encoding =  z_encoding_zenoh_uint32();`?

Reply:
Fixed